agent/rpi_helper.py

```diff
@@ -19,3 +22,10 @@ def detect_raspberry_pi():
                 metadata['serial_number'] = line.split()[-1]
 
     return metadata
+
+
+def get_distro():
```

Comment:
This one is a bit more complicated unfortunately. While this will work when you run it as the python module, it will not work in the Snap or on Balena.

Reply:
Let's do this:

 * Check if the environment variable `SNAP_INSTANCE_NAME` is set. If so, set the distribution to "Snapd" ([reference](https://docs.snapcraft.io/environment-variables/7983))
 * On Balena/Resin, check for the environment variables `RESIN_APP_NAME` or `BALENA_APP_NAME`. If so, set the 'distribution' to 'Balena'